client/app/services/dashboard.js

```diff
@@ -4,6 +4,15 @@ import { Widget } from './widget';
 
 export let Dashboard = null; // eslint-disable-line import/no-mutable-exports
 
+export const urlForDashboard = ({ id, name }) => `dashboard/${id}-${name.toString()
+  .trim()
+  .toLowerCase()
+  .replace(/\s+/g, '-')
+  .replace(/[^\w-]+/g, '')
+  .replace(/--+/g, '-')
+  .replace(/^-+/, '')
+  .replace(/-+$/, '')}`;
```

Comment:
We should still create the slug in the backend, it will just update when you update the dashboard name.

Reply:
Yeah, but that might break existing links, if anyone kept them.

My intention was to have existing slugs continue to work, while new links would be 123-pretty-much-the-name-of-the-dashboard-here-but-could-be-anything-youd-like-cause-we-only-care-about-the-id-from-now.

Reply:
It doesn't have to:

1. We will keep the current `slug` value in the database as is, so we can do lookups for existing dashboards.
2. We will no longer update it, but compute it when returning the API response based on the current name.

Reply:
[name_as_slug](https://github.com/getredash/redash/pull/1009/files#diff-8897279477368a0297f53278e8fdcdd5R871) 😬😬😬